packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -64,6 +64,7 @@ export interface PreferencesState extends BaseState {
   useTokenDetection: boolean;
```

Comment:
We don't use this controller in the extension. Was this change just made in order to support mobile as well?

Reply:
`useTokenDetection` was added for mobile usage at the time, as we were doing the development together.

For `useCurrencyRateCheck`, I added this because I was unsure how to update the rate checking based on preference state change without making corresponding changes here. The PreferenceState type was throwing an error since `useCurrencyRateCheck` was not in the interface definition.